src/auth/services/auth.service.ts

```diff
@@ -599,4 +648,65 @@ export class AuthService {
       }
     }
   }
+  async googleUnlink(userNo: number) {
+    try {
+      const socialTokens = await this.tokenRepository.findToken(userNo);
+      if (!socialTokens) {
+        throw new NotFoundException("user not found");
+      }
+      const user = await this.usersRepository.findUserByUserNo(userNo);
+      if (user.domain !== "google") {
+        throw new UnauthorizedException(
+          "You are not a user logged in with Google.",
+        );
+      }
+      let socialAccessToken = socialTokens.socialAccess;
+      const socialRefreshToken = socialTokens.socialRefresh;
+
+      const socialAccessTokenInfo =
+        await this.tokenService.googleSocialAccessTokenInfo(socialAccessToken);
+      if (socialAccessTokenInfo === 401) {
+        const newGoogleAccessToken =
+          await this.tokenService.createNewGoogleAccessToken(
+            socialRefreshToken,
+          );
+        await this.tokenRepository.updateAccessToken(
+          userNo,
+          newGoogleAccessToken.access_token,
+        );
+        socialAccessToken = newGoogleAccessToken.access_token;
+      }
```

Comment:
이 로직은 어떤 목적인지 알려주실수 있나요?

Reply:
회원 탈퇴에 경우 회사(네이버, 카카오, 구글)에게 만료되지 않은 정상적인 소셜 액세스 토큰을 전달해야 회원탈퇴처리가 됩니다. 그렇기 때문에 회사에게 전달하기 전에 소셜 액세스 토큰을 꺼내와서 정상적인 토큰인지 검증해주고 만약 만료되었다면 소셜 리프레시를 통해 소셜 액세스를 재발급 받아서 회사에게 전달하는 로직입니다 구글을 예로 들면 ex) 회원 탈퇴 하기전 정상적인 소셜 액세스인지 확인 ```suggestion  const socialAccessTokenInfo =
        await this.tokenService.googleSocialAccessTokenInfo(socialAccessToken); ```-> 만약 만료되었다면```suggestion  if (socialAccessTokenInfo === 401) {
        const newGoogleAccessToken =
          await this.tokenService.createNewGoogleAccessToken(
            socialRefreshToken,
          );
 ``` 소셜 리프레시를 통해 소셜 액세스 재발급 후 db상에 소셜 액세스 토큰 업데이트  -> 재발급 받은 소셜 액세스로 회원 탈퇴처리 이런식으로 진행되게 됩니다

Reply:
아하 그래서 재발급을 받는 로직이 있었군요 그 부분 짚어주셔서 감사합니다 😊😊😊😊😊